docs/plot/plot.py

```diff
@@ -1,39 +1,132 @@
+                    than one plotname can be given at once.
+
+Plot definitions:
+----------------
+
+Plots are defined by a JSON dict. A plot is defined by a name, filename,
+bounding box (lat/long space), a proj-string, plottype and resolution.
+An example of the setup for the plot for the Airy projection is shown below:
+
+    {
+        "filename": "airy.png",
+        "latmax": 90,
+        "latmin": -90,
+        "lonmax": 90,
+        "lonmin": -90,
+        "name": "airy",
+        "projstring": "+proj=airy",
+        "res": "low",
+        "type": "poly"
+    }
+
+"res" can be either "low" or "med" and "type" can be either "poly"(gon)
+or "line". The rest of the inputs are fairly free form.
 
-The output dir is optional. Uses current dir if not set.
 
-Change PROJ and COASTLINE_DATA to path on your local system
-before running the script.
+Change PROJ to path on your local system before running the script.
 '''
 
+from __future__ import print_function
+from __future__ import division
+
 import os
+import os.path
 import sys
 import json
 import subprocess
+import functools
 
 import numpy as np
 import matplotlib.pyplot as plt
+from matplotlib.patches import Polygon
+import fiona
+from shapely.geometry import Polygon
+from shapely.geometry import MultiPolygon
+from shapely.geometry import shape
+from shapely.ops import transform
+from descartes import PolygonPatch
+
+PROJ = '../../src/proj'
+#PROJ = 'proj'
+
+LINE_LOW = 'data/coastline.geojson'
+LINE_MED = 'data/coastline50.geojson'
+POLY_LOW = 'data/land.geojson'
+POLY_MED = 'data/land50.geojson'
 
-PROJ = 'proj'
+GRATICULE_WIDTH = 15
+N_POINTS = 1000
 
-COASTLINE_DATA = 'data/coastline.geojson'
+# colors
+COLOR_LAND = '#000000'
+COLOR_COAST = '#000000'
+COLOR_GRAT = '#888888'
 
-GRATICULE_WIDTH = 15
-N_POINTS = 100
+def interp_coords(coords, tol):
+    '''
+    Interpolates extra coordinates when the euclidean distance between adjacent
+    points are larger than given tolerance.
+
+    The function tries to densify a list of coordinates based on a simple measure.
+    If two adjacent points in the input coordinate list are further away from each
+    other than the specified tolerance, the list will be densied between those two
+    points. Roughly speaking, a point will be inserted every `tol` between the
+    points.
+
+    Inputs:
+    -------
+    coords:         2xN numpy array with geodetic coordinates
+    tol:            Euclidean distance tolerance.
 
-LAT_MIN = -90
-LAT_MAX = 90
-LON_MIN = -180
-LON_MAX = 180
+    Returns:
+    --------
+    A densified numpy array of coordinates. If tol < 0 the input array is returned
+    unchanged.
+
+    '''
+    if tol < 0:
+        return coords
+
+    x, y = coords
+    x = np.array(x)
+    y = np.array(y)
+
+    l = len(x)
+    dsts = np.sqrt((x[0:-1]-x[1:l])**2 + (y[0:-1]-y[1:l])**2)
+    I = dsts > tol
+
+    offset = 0
+    xy = np.ndarray(shape=(2, 0))
+    for i, b in enumerate(I):
+        xy = np.append(xy, (x[i], y[i]))
```

Comment:
This is kind of inefficient; it would be better to group things, e.g., make a list where each entry is either a) a bunch of existing points that are below tolerance or b) an interpolated list of points. Then concatenate them at the end. I would try this out, but I can't yet get this script to run.